static/app/components/nav/primary.tsx

```diff
@@ -0,0 +1,332 @@
+import {
+  isNavItemActive,
+  isSubmenuItemActive,
+  makeLinkPropsFromTo,
+  type NavSidebarItem,
+  resolveNavItemTo,
+} from 'sentry/components/nav/utils';
+import {
+  IconDashboard,
+  IconGraph,
+  IconIssues,
+  IconLightning,
+  IconProject,
+  IconQuestion,
+  IconSearch,
+  IconSettings,
+  IconSiren,
+} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import ConfigStore from 'sentry/stores/configStore';
+import {space} from 'sentry/styles/space';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+
+interface SidebarItemProps {
+  item: NavSidebarItem;
+  children?: React.ReactNode;
+  onClick?: MouseEventHandler<HTMLElement>;
+}
+
+function SidebarBody({children}: {children: React.ReactNode}) {
+  return <SidebarItemList>{children}</SidebarItemList>;
+}
+
+function SidebarFooter({children}: {children: React.ReactNode}) {
+  return (
+    <SidebarFooterWrapper>
+      <SidebarItemList>{children}</SidebarItemList>
+    </SidebarFooterWrapper>
+  );
+}
+
+function SidebarItem({item}: SidebarItemProps) {
+  const to = resolveNavItemTo(item);
+  const SidebarChild = to ? SidebarLink : SidebarMenu;
+  const organization = useOrganization();
+
+  const FeatureGuard = item.feature ? Feature : Fragment;
+  const featureGuardProps: any = item.feature ?? {};
+
+  const recordAnalytics = useCallback(
+    () =>
+      trackAnalytics('growth.clicked_sidebar', {item: item.analyticsKey, organization}),
+    [organization, item.analyticsKey]
+  );
+
+  return (
+    <FeatureGuard {...featureGuardProps}>
+      <SidebarItemWrapper>
+        <SidebarChild item={item} key={item.label} onClick={recordAnalytics}>
+          {item.icon}
+          <span>{item.label}</span>
+        </SidebarChild>
+      </SidebarItemWrapper>
+    </FeatureGuard>
+  );
+}
+
+function SidebarMenu({item, children, onClick}: SidebarItemProps) {
+  if (!item.dropdown) {
+    throw new Error(
+      `Nav item "${item.label}" must have either a \`dropdown\` or \`to\` value!`
+    );
+  }
+  return (
+    <DropdownMenu
+      position="right-end"
+      trigger={(props, isOpen) => {
+        return (
+          <NavButton
+            {...props}
+            onClick={event => {
+              onClick?.(event);
+              props.onClick?.(event);
+            }}
+          >
+            <InteractionStateLayer hasSelectedBackground={isOpen} />
+            {children}
+          </NavButton>
+        );
+      }}
+      items={item.dropdown}
+    />
+  );
+}
+
+function SidebarLink({children, item, onClick}: SidebarItemProps) {
+  const location = useLocation();
+  const isActive = isNavItemActive(item, location);
+  const isSubmenuActive = isSubmenuItemActive(item, location);
+  const to = resolveNavItemTo(item);
+  if (!to) {
+    throw new Error(
+      `Nav item "${item.label}" must have either a \`dropdown\` or \`to\` value!`
+    );
+  }
```

Comment:
can this not be done via typescript

Reply:
Yeah, this was existing code. I'm getting rid of the `item` prop anyway, once I'm fully off the config you'll just render a link or dropdown depending on what you want